pkg/api/aim/runs.go

```diff
@@ -413,8 +413,10 @@ func SearchRuns(c *fiber.Ctx) error {
 			ID: q.Offset,
 		}
 		// TODO:DSuhinin -> do we need `namespace` restriction? it seems like yyyyess, but ....
-		if err := database.DB.Select("row_num").First(&run).Error; err != nil && errors.Is(err, gorm.ErrRecordNotFound) {
-			return fmt.Errorf("unable to find search runs offset %q: %w", q.Offset, err)
+		if err := database.DB.Select("row_num").First(&run).Error; err != nil {
+			if !errors.Is(err, gorm.ErrRecordNotFound) {
```

Comment:
hmm, why the extra `if` instead of just `&&` like we do everywhere else?

Reply:
it is more readable.

Reply:
honestly, I don't think it is + it's inconsistent

Reply:
emmm, no. for example in `GetRunInfo`:
```
if err := tx.First(&r).Error; err != nil {
		if tx.Error == gorm.ErrRecordNotFound {
			return fiber.ErrNotFound
		}
		return fmt.Errorf("error retrieving run %q: %w", p.ID, tx.Error)
	}
```

Reply:
`GetApp`:
```
if err := database.DB.
		Where("NOT is_archived").
		First(&app).
		Error; err != nil {
		if err == gorm.ErrRecordNotFound {
			return fiber.ErrNotFound
		}
		return fiber.NewError(fiber.StatusInternalServerError, fmt.Sprintf("unable to find app %q: %s", p.ID, err))
	}
```

Reply:
that's exactly my point, all the places where we do that are places where we return an error when `err != nil` and then specialize the return based on the error type — this is _not_ what we're doing here! we're filtering the errors that are simple "ErrRecordNotFound".

`GetExperimentRuns` (same logic as `SearchRuns` for the limit/offset):

```go
	if q.Offset != "" {
		run := &database.Run{
			ID: q.Offset,
		}
		if err = database.DB.Select("row_num").First(&run).Error; err != nil && err != gorm.ErrRecordNotFound {
			return fmt.Errorf("unable to find search runs offset %q: %w", q.Offset, err)
		}

		tx.Where("row_num < ?", run.RowNum)
	}
```

Reply:
ok. I disagree but reverted it back.